test/unit/connection_string.test.ts

```diff
@@ -89,6 +94,39 @@ describe('Connection String', function () {
     expect(options.credentials.source).to.equal('$external');
   });
 
+  it('should retain user-specified authSource with no credentials for non-srv URI', function () {
+    const mockAuthSource = 'thisShouldBeAuthSource';
+    const options = parseOptions(`mongodb://localhost/?authSource=${mockAuthSource}`);
+
+    expect(options).to.have.nested.property('credentials.username', '');
+    expect(options).to.have.nested.property('credentials.mechanism', 'DEFAULT');
+    expect(options).to.have.nested.property('credentials.source', mockAuthSource);
+  });
+
+  it('should omit credentials and throw a non-MongoAPIError if the only auth related option is authSource', async () => {
```

Comment:
```suggestion
  it('should omit credentials and not throw a MongoAPIError if the only auth related option is authSource', async () => {
```

Reply:
Yea I'm relying on no server running on localhost:123, and then checking we're not throwing the InvalidArg error, so I agree with the phrasing update here.